### PR TITLE
Proposal: Zero-Knowledge Soulbond Token

### PR DESCRIPTION
**Pull request by wuyahuang**
#### Project Abstract
The zkSBT (Zero-Knowledge Soulbond Token) is an enhanced Soulbond Token that uses ZKPs to provide security and privacy features compared to traditional SBTs. Similar to SBTs, zkSBTs are easy to mint and use for verification purposes, offering the same user experience but with added privacy. The users can easily prove their specific identities through zkSBT without inputting any private key to sign the signatures.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Hi @wuyahuang thanks for the application. The technology definitely seems interesting. A couple of initial questions:

- The [team repo](https://github.com/As-Match/asMatch-front-end) you provided appears to be private.
- Can you add more technical details to the delivery tables? For example in Milestone 1, what framework with the front-end use? What language(s) will be used (other than ink!)
- How is this project related to the [Manta NPO](https://mantanetwork.medium.com/introducing-zksbts-zknfts-and-npo-89a8e6d77a7f) platform? It seems as though they already announced a decentralized [platform](https://npo.manta.network/calamari/sbt) with an accompanying [SDK](https://github.com/Manta-Network/sdk/blob/fer-sbt-ledger/manta-js/package/README_SBT.MD#sbt) that allows users to mint zkSBTs?
- It seems you have [already built](https://galxe.com/credential/273062317152444416) an initial asMatch iOS app in Test Flight? Is that correct?

File: applications/zkSBT.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+* AIGC Profiles - Generate unique profiles using AIGC technology
+* On-chain Credentials - Privately authorize personal information and token holder verification
+* ZK Verification - Privately verify ownership of assets without revealing the associated wallet address & Embark on a Web3 journey without the need to connect wallets
+
+The following is the workflow designed for our zkSBT service for Polkadot.
+
+For AsMatch Team:
+
+* Deploy ink SBT smart contract to Polkadot blockchain
```
Comment (keeganquigley) [inline applications/zkSBT.md:31]: Polkadot is just a relay chain and therefore doesn't support smart contract deployment. Did you mean to say deploy to Manta here?

File: applications/zkSBT.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+* Education history - When people graduate from university, they receive a certificate proving completion of the required courses. The university could be a Soul issuing the SBTs, and the students would be Souls on the receiving end. The SBT would store the student’s credentials, proving they hold the relevant qualifications and are a member of the university. Simply put, the SBT would function as proof of attendance.
+
+* Job applications – In theory, job applicants could submit all their prior work history and professional certificates using official SBTs issued by previous companies and institutions. The SBTs would function as proof of skill certificates.
+
+After all, the zkSBT that we are proposing is not only a use case for AsMatch but also a best SBT practice for the Polkadot ecosystem. The ink SBT smart contract we are developing is a good example for Dapps which also want to deploy SBT on the Polkadot blockchain. And we are ready to integrate more SBTs from the Polkadot blockchain into AsMatch users.
```
Comment (keeganquigley) [inline applications/zkSBT.md:66]: Just to reiterate, ink! contracts can't be deployed to Polkadot, and would have to be deployed to a parachain. But I'm not sure if Manta supports ink! or not yet at this point.

**Comment by wuyahuang** [comment]
Hi @keeganquigley, thanks for your reply.

1. The code repository is not open source for now, but I can grant access to your GitHub account. Is that ok?
2. I already added info on the developing framework and language to the delivery tables.
3. We are going to use the SDK from the Manta NPO platform to generate the ZK proof key.
4. The AsMatch is already published to Google Play and iOS Testflight, including matching and chatting features. We hope the zkSBT will enable users to prove their specific credentials.

**Comment by keeganquigley** [comment]
Thanks @wuyahuang no need to give me access. That's fine for now, but just be aware that upon milestone delivery, we expect the code to be published under the license listed in the application. You can find more on this [here](https://github.com/w3f/Grants-Program#guidelines).

Regarding the mobile apps, please note that we only fund software that doesn't exist yet. Therefore perhaps you could remove this milestone and drop the price to a level 2 proof of concept, and then potentially go for a follow-up grant afterwards. We usually prefer this for teams we haven't worked with before. Feel free to argue against it.

However, could you also please add what functions would be included in the ink! smart contract?

**Comment by wuyahuang** [comment]
Hi @keeganquigley, thanks for your reply.

* Yep, All delivery source codes will be under Apache 2.0 license as we mentioned in the delivery tables.
* Actually, AsMatch is not an *exist* blockchain application, but still a web2 APP without integrating any blockchain. I strongly recommended the Polkadot ecosystem when the dev team is at the stage of blockchain research, that's the reason why we are here. And the AsMatch app is not connecting to any web3 wallet, we are still thinking to connect which chain, if we get the grant we can connect the wallet on Polkadot for this app in the future version.
* I already added our planned ink! functions into the Project Details part and Milestone 3.

**Comment by semuelle** [comment]
Thank you for the application, @wuyahuang, but I don't think this is a good fit for the grants program. We look for contributions that are useful for the ecosystem as a whole, and I don't think your project garners sufficiently broad interest. Effectively, you are then asking for 50,000 USD to implement a smart contract that already exists and a backend service that syncs data between two parachains. 

There might be a use case for enabling zkSBTs on other parachains, but a grant for this would have to contain an airtight architecture and technical roadmap. Therefore, I am going to close this application.

**Pull request closed without merging**